### ISSUE TITLE
Troubles reading xarray netcdf files

### ISSUE THREAD

**Issue by NotSqrt** [georust/netcdf#79]
Hello !

I'm trying the examples/ncdump.rs bin to read files generated from python by `xarray.Dataset.to_netcdf` : 
https://xarray.pydata.org/en/stable/generated/xarray.Dataset.to_netcdf.html

https://github.com/georust/netcdf/blob/ca0970edf8f28517319b0a08b010eba449c6ec94/examples/ncdump.rs#L71
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Netcdf(-45)', netcdf-0.5.2/src/variable.rs:138:56
```

The `-45` being `NC_EBADTYPE` from https://github.com/georust/netcdf/blob/71b41fe/netcdf-sys/src/netcdf_const.rs#L109, I suppose.

I added a catch_unwind to get more details: it seems to fail on types coming from numpy: fixed-length sequence of chars (`dtype('S1')`, `dtype('S4')`, etc.)

Any idea on where to go from there ?

Thanks a lot !

**Comment by magnusuMET** [comment]
`NC_EBADTYPE` is an error from `netcdf-c`. I think `ncdump` might do something wrong regarding printing the type (the `vartype()` call). Do you have an example file that shows this problem? Does the `ncdump` which uses `netcdf-c` print the type correctly?

**Comment by magnusuMET** [comment]
Looks like `S1` and `S4` are converted to `NC_CHAR` which we did not support in this library. This should be fixed in the linked PR.

Do note that `NC_CHAR` is awkward. One cannot read into `u8` or `i8` (`nc_get_vara_{u,s}char`) directly, but must use `nc_get_vara`. Our wrapper of `nc_get_vara` is called `raw_values` and returns raw bytes the user must interpret.

**Comment by NotSqrt** [comment]
I confirm that `netcdf = { git = "https://github.com/magnusuMET/netcdf", branch = "feature/char" }` fixes the panic !

To give a complete example: a matrix of min/average/max RMS values over 2 signals x 15 times:
```python
import xarray
import pandas
import numpy

ds = xarray.Dataset(
    {"rms": (['f', 'time', 'quantity'], numpy.random.rand(3, 15, 2))},
    coords={
        'f': [b'min', b'mean', b'max'],
        'time': pandas.date_range("2021-01-01", periods=15, freq='60S'),
        'quantity': [b'A', b'B'],
    }
)
ds.to_netcdf('/tmp/example.nc')

print(ds)
```
shows:
```
<xarray.Dataset>
Dimensions:   (f: 3, quantity: 2, time: 15)
Coordinates:
  * f         (f) |S4 b'min' b'mean' b'max'
  * time      (time) datetime64[ns] 2021-01-01 ... 2021-01-01T00:14:00
  * quantity  (quantity) |S1 b'A' b'B'
Data variables:
    rms       (f, time, quantity) float64 0.3632 0.8429 0.5622 ... 0.3649 0.3538
```

while ncdump shows:
```
Dimensions:
        f : (3)
        time : (15)
        quantity : (2)
        string4 : (4)
        string1 : (1)
Variables:
        rms( f  time  quantity ): f64
                _FillValue = Double(NaN)
        f( f  string4 ): char
        time( time ): i64
                units = Str("minutes since 2021-01-01 00:00:00")
                calendar = Str("proleptic_gregorian")
        quantity( quantity  string1 ): char
```

**I'm suspicious of the `string4` and `string1` shown in the dimensions, but it might be the way xarray encodes its data.**

**Comment by magnusuMET** [comment]
`string4` and `string1` are automatically created names for unnamed dimensions in `xarray`. You might find it easier to work with strings as you already have some string items. The following stack overflow questions has some details regarding automatic conversion: https://stackoverflow.com/questions/48959201/xarray-store-strings-as-string-data-type-instead-of-char-n-dimensional-ch

There is also some documentation in `xarray` on these conversions: http://xarray.pydata.org/en/stable/io.html#string-encoding.

**Comment by magnusuMET** [comment]
Should be available as `0.6.0` on `crates.io` now!
